### DISCUSSION TITLE
Updates to Fanuc Agent

### DISCUSSION THREAD

**Discussion by PatrickRitchie** [TrakHound/Fanuc-MTConnect-Agent#12] [Announcements]
I'm looking to update this repo and to rewrite the Fanuc MTConnect Adapter in .NET in order to use our new [MTConnect.NET](https://github.com/TrakHound/MTConnect.NET) Agent and Library.

If anyone has any comments or feature requests then please feel free to leave a comment.

**Comment by kidadoor** [comment]
Hi @PatrickRitchie, 

Thank you for making and maintaining this thread. I am using the old Fanuc-MTConnect-Agent and collecting data from Fanuc controller over ethernet form some time. I remember that I used to receive all the parameters but I see that suddenly I have stopped receiving ToolID from all the controllers. Do you think it might be the software issue or settings in the controller itself have been changed?

Thanks and regards

**Reply by smtpl-iiotian** [reply]
@kidadoor, I faced the same issue for toolid but after restarting the agent and adapter toolid data was resumed.

**Comment by smtpl-iiotian** [comment]
Read Macros in Fanuc-MTConnect-Agent

@PatrickRitchie,  I updated adapter.ini for each individual adapter as

[macros]
[cycletime = 803]

and devices.xml file with Component Path stream as

DataItem category="EVENT" id="dev_01_cycletime_01" name="cycletime" type="CYCLETIME"

cycletime variable not seen in agent.log. UNAVAILABLE data appears on http://localhost:5000/current.  

Kindly guide.

**Comment by Mancio3096** [comment]
Signature verification
A certificate was explicitly revoked by its issuer.

I can't install software on windows. How can i do?
